src/osd/OSD.cc

```diff
@@ -4041,6 +4047,69 @@ void OSD::load_pgs()
 }
 
 
+PGRef OSD::handle_pg_create_info(OSDMapRef osdmap, const PGCreateInfo *info)
+{
+  spg_t pgid = info->pgid;
+
+  int up_primary, acting_primary;
+  vector<int> up, acting;
+  osdmap->pg_to_up_acting_osds(
+    pgid.pgid, &up, &up_primary, &acting, &acting_primary);
+
```

Comment:
We create the PG in the epoch the pool was created in, so it walks forward in time and builds up history as a PG (instead of via a special implementation in the form of build_initial_history()).

Reply:
Sounds good.